test/credentials_test.dart

```diff
@@ -61,36 +61,99 @@ Future<void> main([List<String>? args]) async {
     expect(user, isNotNull);
   });
 
+  baasTest('Email/Password - register user auto confirm', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "${generateRandomString(5)}@bar.com";
+    String password = "SWV23R#@T#VFQDV";
+    // For application with name 'auto' and with confirmationType = 'autoConfirm'
+    // all the usernames are automatically confirmed.
+    await authProvider.registerUser(username, password);
+  }, appName: "autoConfirm");
+
   baasTest('Email/Password - register user twice throws', (configuration) async {
     final app = App(configuration);
     final authProvider = EmailPasswordAuthProvider(app);
-    String emailPrefix = generateRandomString(5);
-    String username = "realm_tests_do_autoverify$emailPrefix@bar.com";
+    String username = "${generateRandomString(5)}@bar.com";
     String password = "SWV23R#@T#VFQDV";
     await authProvider.registerUser(username, password);
     expect(() async {
       await authProvider.registerUser(username, password);
     }, throws<RealmException>("name already in use"));
-  });
+  }, appName: "autoConfirm");
 
   baasTest('Email/Password - register user with weak/empty password throws', (configuration) async {
     final app = App(configuration);
     final authProvider = EmailPasswordAuthProvider(app);
-    String emailPrefix = generateRandomString(5);
-    String username = "realm_tests_do_autoverify$emailPrefix@bar.com";
+    String username = "${generateRandomString(5)}@bar.com";
     expect(() async {
       await authProvider.registerUser(username, "pwd");
     }, throws<RealmException>("password must be between 6 and 128 characters"));
     expect(() async {
       await authProvider.registerUser(username, "");
     }, throws<RealmException>("password must be between 6 and 128 characters"));
-  });
+  }, appName: "autoConfirm");
 
   baasTest('Email/Password - register user with empty email throws', (configuration) async {
     final app = App(configuration);
     final authProvider = EmailPasswordAuthProvider(app);
     expect(() async {
       await authProvider.registerUser("", "password");
     }, throws<RealmException>("email invalid"));
+  }, appName: "autoConfirm");
+
+  baasTest('Email/Password - confirm user token expired', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "${generateRandomString(5)}@hotmail.com";
+    String password = "SWV23R#@T#VFQDV";
+    await authProvider.registerUser(username, password);
+    expect(() async {
+      await authProvider.confirmUser(
+          "0e6340a446e68fe02a1af1b53c34d5f630b601ebf807d73d10a7fed5c2e996d87d04a683030377ac6058824d8555b24c1417de79019b40f1299aada7ef37fddc",
+          "6268f7dd73fafea76b730fc9");
```

Comment:
How are these tokens generated?

Reply:
They were sent to my email after registration. I just copied them. They are old tokens, but in a valid format. In this case the validation is throwing different exception.

Reply:
Hm... I may be wrong, but I don't think we'll be able to use them like that. These are just strings and not JWTs that encode their expiration - I believe they're just stored in the database, so when you try to run this test against a different app, it'll fail with invalid token rather than expired token. Once we fix test runs on CI, I expect this test to fail, but let's see.

Reply:
Yes, right, but they are running on the same app in this manual test. It would be good if I had the server code to check the validation, but I couldn't access it.

Reply:
This test is not manual. We are passing expired tokens and the exception is thrown.

Reply:
ok thanks